content/en/blog/plugins-early-access.md

```diff
@@ -0,0 +1,157 @@
+
+#### Plugin Events In Capture Files
+
+Plugin events are compatible with `.scap` capture files and can be read by Falco and read/written by userspace tools like [sysdig](https://github.com/draios/sysdig) that use the Falco libraries. A new event type "pluginevent" contains a plugin id and an arbitrary payload returned by the plugin. This allows leveraging existing support for capture files in the Falco libraries. A capture file can potentially contain a mix of system call and plugin events in the same file. You can even read capture files generated by plugins in tools like [wireshark](https://gitlab.com/wireshark/wireshark)!
+
+### Plugin Event Sources and Interoperability
+
+With the introduction of plugins, there's the potential for many new kinds of events and many new fields that can extract information from events. With that in mind, we've formalized how rules are matched to events in Falco.
+
+Events returned by source plugins have an "event source" which describes the information in the event. This is distinct from the plugin name to allow for multiple kinds of plugins to generate the same kind of events. For example, there might be plugins gke-audit-bridge, eks-audit-bridge, ibmcloud-audit-bridge, etc. that all fetch K8s Audit information. The plugins would be distinct, with their own shared libraries and implementations, but would have the same event source "k8s_audit".
+
+When Falco loads rules, it reads the `source:` property and tries to match that event source to a loaded source and/or extractor plugin. If a source plugin is loaded that supports that event source, all rules with that source will be evaluated for events coming from that source plugin.
+
+Extractor plugins are handled slightly differently. An extractor plugin optionally provides a set of compatible event sources. When the framework receives an event with an event source that is in the compatible event sources list, fields in expressions/Falco outputs will be extracted from events using the extractor plugin. An extractor plugin can also not name a set of event sources. In this case, all events will be presented to the extractor plugin, regardless of source. In this case, the extractor plugin must detect the format of arbitrary payloads and be able to return NULL/no value when the payload is not supported.
+
+### Configuring Plugins in Falco
+
+Plugins are configured in yaml via new `falco.yaml` properties `plugins` and `load_plugins`. Here's an example:
+
+```yaml
+plugins:
+  - name: cloudtrail
+    library_path: cloudtrail/plugin.so
+    init_config: "..."
+    open_params: "..."
+  - name: json
+    library_path: json/plugin.so
+    init_config: ""
+    open_params: ""
+
+# Optional
+load_plugins: [cloudtrail, http_json]
+```    
+
+The `plugins` property in falco.yaml defines the set of plugins that can be loaded by Falco. It contains a list of plugin definitions. Each plugin definition has:
```

Comment:
The `plugins` property in `falco.yaml` defines the set of plugins that can be loaded by Falco. The property contains a list of plugin definitions. Each plugin definition has: